reporter/internal/pdata/generate.go

```diff
@@ -0,0 +1,271 @@
+// this moment.
+func (p *Pdata) setProfile(
+	events map[samples.TraceAndMetaKey]*samples.TraceEvents,
+	profile pprofile.Profile,
+) {
+	// stringMap is a temporary helper that will build the StringTable.
+	// By specification, the first element should be empty.
+	stringMap := make(map[string]int32)
+	stringMap[""] = 0
+
+	// funcMap is a temporary helper that will build the Function array
+	// in profile and make sure information is deduplicated.
+	funcMap := make(map[samples.FuncInfo]int32)
+	funcMap[samples.FuncInfo{Name: "", FileName: ""}] = 0
+
+	st := profile.SampleType().AppendEmpty()
+	st.SetTypeStrindex(getStringMapIndex(stringMap, "samples"))
+	st.SetUnitStrindex(getStringMapIndex(stringMap, "count"))
+
+	pt := profile.PeriodType()
+	pt.SetTypeStrindex(getStringMapIndex(stringMap, "cpu"))
+	pt.SetUnitStrindex(getStringMapIndex(stringMap, "nanoseconds"))
+	profile.SetPeriod(1e9 / int64(p.samplesPerSecond))
+
+	// Temporary lookup to reference existing Mappings.
+	fileIDtoMapping := make(map[libpf.FileID]int32)
+
+	attrMgr := samples.NewAttrTableManager(profile.AttributeTable())
+	var locationIndex int32
+	var startTS, endTS pcommon.Timestamp
+	for traceKey, traceInfo := range events {
+		sample := profile.Sample().AppendEmpty()
+		sample.SetLocationsStartIndex(locationIndex)
+
+		slices.Sort(traceInfo.Timestamps)
+		startTS = pcommon.Timestamp(traceInfo.Timestamps[0])
+		endTS = pcommon.Timestamp(traceInfo.Timestamps[len(traceInfo.Timestamps)-1])
+
+		sample.TimestampsUnixNano().FromRaw(traceInfo.Timestamps)
+		sample.Value().Append(1)
+
+		// Walk every frame of the trace.
+		for i := range traceInfo.FrameTypes {
+			loc := profile.LocationTable().AppendEmpty()
+			loc.SetAddress(uint64(traceInfo.Linenos[i]))
+			attrMgr.AppendOptionalString(loc.AttributeIndices(),
+				"profile.frame.type", traceInfo.FrameTypes[i].String())
+
+			switch frameKind := traceInfo.FrameTypes[i]; frameKind {
+			case libpf.NativeFrame:
+				// As native frames are resolved in the backend, we use Mapping to
+				// report these frames.
+
+				var locationMappingIndex int32
+				if tmpMappingIndex, exists := fileIDtoMapping[traceInfo.Files[i]]; exists {
+					locationMappingIndex = tmpMappingIndex
+				} else {
+					idx := int32(len(fileIDtoMapping))
+					fileIDtoMapping[traceInfo.Files[i]] = idx
+					locationMappingIndex = idx
+
+					ei, exists := p.Executables.GetAndRefresh(traceInfo.Files[i],
+						ExecutableCacheLifetime)
+
+					// Next step: Select a proper default value,
+					// if the name of the executable is not known yet.
+					var fileName = "UNKNOWN"
+					if exists {
+						fileName = ei.FileName
+					}
+
+					mapping := profile.MappingTable().AppendEmpty()
+					mapping.SetMemoryStart(uint64(traceInfo.MappingStarts[i]))
+					mapping.SetMemoryLimit(uint64(traceInfo.MappingEnds[i]))
+					mapping.SetFileOffset(traceInfo.MappingFileOffsets[i])
+					mapping.SetFilenameStrindex(getStringMapIndex(stringMap, fileName))
```

Comment:
I'm wondering where this function is coming from and why it is called the way it is. `Strindex` should be `StrIndex`.
```suggestion
					mapping.SetFilenameStrIndex(getStringMapIndex(stringMap, fileName))
```

Reply:
It comes from the pprofile package.
https://pkg.go.dev/go.opentelemetry.io/collector/pdata/pprofile#Function.SetFilenameStrindex

That code is auto-generated from configuration to match the proto fields.
https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/cmd/pdatagen/internal/pprofile_package.go#L361
We could rename it in the code generator. For a full match, we'd have to rename the proto field as well though.